pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -1134,26 +1143,31 @@ func (re registrationEntries) Less(i, j int) bool {
 	return false
 }
 
-func (ds *sqlitePlugin) sortEntries(entries []*common.RegistrationEntry) []*common.RegistrationEntry {
+func (ds *sqlPlugin) sortEntries(entries []*common.RegistrationEntry) []*common.RegistrationEntry {
 	e := registrationEntries(entries)
 	sort.Sort(e)
 	return []*common.RegistrationEntry(e)
 }
 
-// restart will close and re-open the sqlite database.
-func (ds *sqlitePlugin) restart() error {
+// restart will close and re-open the gorm database.
+func (ds *sqlPlugin) restart() error {
 	ds.mutex.Lock()
 	defer ds.mutex.Unlock()
 
-	// Build sqlite connect string
-	path := ds.fileName
-	if path == ":memory:" {
-		path = path + "?cache=shared"
+	var db *gorm.DB
+	var err error
+
+	switch ds.DatabaseType {
+	case "sqlite3":
+		db, err = sqlite{}.connect(ds.ConnectionString)
+	case "postgres":
+		db, err = postgres{}.connect(ds.ConnectionString)
+	case "mysql":
+		db, err = mysql{}.connect(ds.ConnectionString)
+	default:
+		err = fmt.Errorf("Unsupported database_type: %v", ds.DatabaseType)
```

Comment:
nit: lowercased error string is preferred. Perhaps we can also simply return here since we know we're going to fail?

Reply:
Applied your requested changes.